### PR TITLE
Adding Interactive Initialization Documentation

### PR DESCRIPTION
**Pull request by smithis7**
This PR creates interactive documentation on energy packet initialization.

**Description**

The new notebook incorporates the "Discretization" page and elements of the "Propagation" page of the documentation, as well as code that demonstrates how packets are initialized in TARDIS. This PR also fixes hyperlink problems that arose from my changes.

**Motivation and context**

This is part of a larger problem to make the TARDIS documentation more interactive.

**How has this been tested?**
- [ ] Testing pipeline.
- [x] Other. 
The documentation has been built locally and on github (see below)

**Examples**

You can view the built documentation with my changes [here](https://smithis7.github.io/tardis/branch/initialization_doc/physics/montecarlo/initialization.html).

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [x] I have updated the documentation accordingly.
    - [x] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: docs/physics/montecarlo/initialization.ipynb [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    "\n",
+    "Finally, each packet is assigned an initial frequency (or more precisely, the initial frequency of its consitiuent photons). Note that since each packet has the same energy, each packet will represent a different number of real photons. The sampling on packet frequencies is more involved than that of the propagation direction, as it involves sampling the Planck distribution (see below). TARDIS uses the technique described in <strong data-cite=\"Carter1975\">[]</strong> and summarized in <strong data-cite=\"Bjorkman2001\">[]</strong> for this purpose.\n",
+    "\n",
+    "During the simulation, the energy of the packet remains constant in the local\n",
+    "co-moving frame (see [Reference Frames](propagation.rst#reference-frames)). This naturally ensures energy\n",
+    "conservation and constitutes the main advantage of this discretization scheme. **However, while the energy of the packets is conserved in the co-moving frame, the frequency of the contituent photons (in the local co-moving frame) may over the course of the simulation. Thus, a packet may represent several different numbers of real photons throughout their lifetimes.**\n",
```
Comment (andrewfullard) [inline docs/physics/montecarlo/initialization.ipynb:49]: may *vary*

File: docs/physics/montecarlo/initialization.ipynb [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    "#The random number generator that will be used\n",
+    "rng = np.random.default_rng()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ae02998",
+   "metadata": {},
+   "source": [
+    "The followig cell contains values that you can change to see how it affects the spectrum:"
```
Comment (andrewfullard) [inline docs/physics/montecarlo/initialization.ipynb:75]: following

File: docs/physics/montecarlo/initialization.ipynb [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    "r_phot = 1e15 #radius of the supernova's photosphere in cm\n",
+    "n_packets = 40000 #number of packets generated"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "450faf76",
+   "metadata": {},
+   "source": [
+    "You can either set a temperatature of the photosphere, which will determine its luminosity; or you can set the luminosity of the photosphere, which will determine it's temperature."
```
Comment (andrewfullard) [inline docs/physics/montecarlo/initialization.ipynb:95]: its

File: docs/physics/montecarlo/initialization.ipynb [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    "\n",
+    "lumin_per_packet = energies/t_simulation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d839222",
+   "metadata": {},
+   "source": [
+    "We define important constants, and for comparisson's sake, we code the Planck distribution function\n",
```
Comment (andrewfullard) [inline docs/physics/montecarlo/initialization.ipynb:167]: comparison's

**Review by andrewfullard — COMMENTED** [review]
Looks great just needs a few typos fixed

**Comment by smithis7** [comment]
> Looks great just needs a few typos fixed

Fixed. I really need to put my writing into a spelling/grammar check program before I make a PR *insert facepalm emoji*

**Comment by smithis7** [comment]
Converted to draft until #1592 is merged so I can edit this notebook to reflect the changes to the code.

**Review by andrewfullard — APPROVED** [review]

File: docs/physics/montecarlo/initialization.ipynb [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    "\n",
+    "Finally, each packet is assigned an initial frequency (or more precisely, the initial frequency of its consitiuent photons). Note that since each packet has the same energy, each packet will represent a different number of real photons. The sampling on packet frequencies is more involved than that of the propagation direction, as it involves sampling the Planck distribution (see below). TARDIS uses the technique described in <strong data-cite=\"Carter1975\">[]</strong> and summarized in <strong data-cite=\"Bjorkman2001\">[]</strong> for this purpose.\n",
+    "\n",
+    "During the simulation, the energy of the packet remains constant in the local\n",
+    "co-moving frame (see [Reference Frames](propagation.rst#reference-frames)). This naturally ensures energy\n",
+    "conservation and constitutes the main advantage of this discretization scheme. **However, while the energy of the packets is conserved in the co-moving frame, the frequency of the contituent photons (in the local co-moving frame) may vary over the course of the simulation. Thus, a packet may represent several different numbers of real photons throughout their lifetimes.**\n",
```
Comment (chvogl) [inline docs/physics/montecarlo/initialization.ipynb:49]:
```suggestion
    "conservation and constitutes the main advantage of this discretization scheme. **However, while the energy of the packets is conserved in the co-moving frame, the frequency of the constituent photons (in the local co-moving frame) may vary over the course of the simulation. Thus, a packet may represent several different numbers of real photons throughout their lifetimes.**\n",
```

File: docs/physics/montecarlo/initialization.ipynb [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "426325e5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from tardis.montecarlo.packet_source import *\n",
```
Comment (chvogl) [inline docs/physics/montecarlo/initialization.ipynb:62]: I would avoid a star import. Just import the `BlackBodySimpleSource` (and everything else you need explicitly).

File: docs/physics/montecarlo/initialization.ipynb [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc34bf33",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "seed = 1 #seed for the pseudo-random number generator\n",
+    "r_phot = 1e15 #radius of the supernova's photosphere in cm\n",
+    "n_packets = 40000 #number of packets generated"
```
Comment (chvogl) [inline docs/physics/montecarlo/initialization.ipynb:87]: This cell and the next few code cells have some minor PEP 8 violations. In this cell, the inline comments do not follow the guidelines: "Inline comments should be separated by at least two spaces from the statement. They should start with a # and a single space."
Maybe reformat this cell and the next few using black outside of the notebook?

**Review by chvogl — CHANGES_REQUESTED** [review]
This looks great! Really nice job writing up a complex topic. My only concern are some minor PEP 8 issues, which would be good to fix.

**Comment by Rodot-** [comment]
Try to use astropy units when instantiating quantities containing units to keep consistency with the way that constants are access from astropy.

**Review by Rodot- — APPROVED** [review]

**Pull request merged**